deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/Hdf5Archive.java

```diff
@@ -61,6 +61,21 @@ public Hdf5Archive(String archiveFilename) {
         this.file = new hdf5.H5File(archiveFilename, H5F_ACC_RDONLY);
     }
 
+    private hdf5.Group[] openGroups(String ... groups) {
```

Comment:
Neat. How do you see the reference pattern to this array evolve? If it's always the last element being accessed, could we make it a stack and save ourselves the potential `OutOfBounds` issues?

Reply:
Don't know, it hasn't really evolved, maybe that's all we'll ever need from HDF5, @maxpumperla ? @turambar ?